modules/nodes-custom-scheduler-packaging.adoc

```diff
@@ -0,0 +1,112 @@
+// Module included in the following assemblies:
+//
+// * nodes/scheduling/nodes-custom-scheduler.adoc
+
+[id="nodes-custom-scheduler-packaging_{context}"]
+= Packaging and deploying a custom scheduler
+
+To include a custom scheduler in your cluster, create an image and include that image in a deployment.
+
+.Prerequisites
+
+* You have access to the cluster as a user with the `cluster-admin` role.
+* You have created a scheduler binary.
++
+[NOTE]
+====
+Information on how to create the scheduler binary is outside the scope of this document. For an example, see link:https://kubernetes.io/docs/tasks/administer-cluster/configure-multiple-schedulers[Configure Multiple Schedulers] in the Kubernetes documentation.
+====
+
+.Procedure
+
+. Create a container image containing the scheduler binary.
+
+.. Create a Dockerfile:
++
+----
+FROM <source_image>
+ADD <path_to_binary> /usr/local/bin/kube-scheduler
+----
+
+.. From the directory that contains the Dockerfile, build the image:
++
+----
+$ docker build -t <dest_env_registry_ip>:<port>/<namespace>/<image_name>:<tag>
+----
+
+.. From the directory that contains the Dockerfile, push the image to a registry:
++
+----
+$ docker push <dest_env_registry_ip>:<port>/<namespace>/<image_name>:<tag>
+----
+
+. Deploy the custom scheduler.
+
+.. Create a file that contains the deployment resources for the custom scheduler:
++
+.Example `custom-scheduler.yaml` file
+[source,yaml]
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: custom-scheduler
+  namespace: kube-system <1>
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: custom-scheduler
+subjects:
+- kind: ServiceAccount
+  name: custom-scheduler
+  namespace: kube-system <1>
+roleRef:
+  kind: ClusterRole
+  name: system:kube-scheduler
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: custom-scheduler
+  namespace: kube-system <1>
+  labels:
+    app: custom-scheduler
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: custom-scheduler
+  template:
+    metadata:
+      labels:
+        app: custom-scheduler
+    spec:
+      serviceAccount: custom-scheduler
+      containers:
+        - name: custom-scheduler
+          image: "<namespace>/<image_name>:<tag>" <2>
+          imagePullPolicy: Always
+----
```

Comment:
I just copy pasted this into a file and when ran with 4.5 cluster i saw the custom-scheduler went into crashloopbackoff state. Can we change the yaml file to below as we generally use this for testing.

[1] https://github.com/openshift/cucushift/blob/master/testdata/customscheduler/my-scheduler-4.6.yaml

Reply:
I updated to use the YAML in the linked file. I updated to change the instances of "my-scheduler" to "custom-scheduler" to be consistent with what the rest of the doc procedure expects. I also updated the `image` field to use use replaceable values of `"<namespace>/<image_name>:<tag>"`.

Here is the diffs between the two YAML files after these changes:
https://github.com/openshift/openshift-docs/compare/4a32d9282acb2956368419fda344f18cc57aa2bd..15c6b3a651daff05401f768ba0d080511cf61279

It adds a lot of additional config to the deployment object. I wanted to check whether all of these updates are necessary. And if they are, do we want to add callouts to additional information about any of these new fields?